cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/Comdb2Handle.java

```diff
@@ -1624,8 +1635,17 @@ private int convert_rc(int rc) {
     }
 
     @Override
-    public synchronized int next() {
+    public int next() {
+        lock.lock();
 
+        try {
+            return lockedNext();
```

Comment:
Let me know if you have an alternative naming suggestion for the private methods that are executed under their respective locks.